cmd/exec.go

```diff
@@ -58,22 +70,109 @@ var nmap = &cobra.Command{
 	Use:   "nmap",
 	Short: "execute nmap",
 	Long:  `executes nmap and splits up the job between all of the specified hosts returning the xml files to the specified directory`,
+	Args: func(cmd *cobra.Command, args []string) error {
+		_, err := deployer.ValidatePorts(nmapPorts)
+		if err != nil {
+			return err
+		}
+		_, err = deployer.ParseIPFile(nmapHostFile)
+		if err != nil {
+			return err
+		}
+
+		marshalledState := deployer.TerraformStateMarshaller()
+
+		list := deployer.ListIPAddresses(marshalledState)
+		if !deployer.IsValidNumberInput(nmapIndex) {
+			return fmt.Errorf("invalid formatting specified: %s", nmapIndex)
+		}
+		listOfNums := deployer.ExpandNumberInput(numberInput)
+
+		largestInstanceNum := deployer.FindLargestNumber(listOfNums)
+
+		//make sure the largestInstanceNumToDestroy is not bigger than totalInstancesAvailable
+		if len(list) < largestInstanceNum {
+			return errors.New("the number you entered is too big. Try running `list` to see the number of instances you have")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		apps := []string{"nmap", "nmap-exec"}
+
+		playbook := deployer.GeneratePlaybookFile(apps)
+
+		numsToDeploy := deployer.ExpandNumberInput(nmapIndex)
+
+		marshalledState := deployer.TerraformStateMarshaller()
+
+		list := deployer.ListIPAddresses(marshalledState)
+
+		var instances []deployer.ListStruct
+
+		for _, num := range numsToDeploy {
+			instances = append(instances, list[num])
+		}
+
+		nmapCommands := deployer.SplitNmapCommand(nmapPorts, nmapHostFile, nmapCommand, len(instances), nmapEvasive)
```

Comment:
what does `SpliNmapCommand` do? probably need to consider a better name

Reply:
Changed to SplitNmapCommandsIntoHosts